### DISCUSSION TITLE
Support bfloat16 dtype in model builder

### DISCUSSION THREAD

**Discussion by BowenBao** [microsoft/onnxruntime-genai#691] [New features / APIs]
Hi team, is there plan to support building bfloat16 models from model builder? cc @kunal-vaishnavi

**Comment by kunal-vaishnavi** [comment]
Yes, we want to support building `bfloat16` models from the model builder. We will need to revisit how the weights are saved because they are currently converted to NumPy arrays and NumPy doesn't support `bfloat16` precision currently.

https://github.com/microsoft/onnxruntime-genai/blob/e41fb2c1430693157c3cea70d3bdcc207ae84ef3/src/python/py/models/builder.py#L406-L417

**Reply by BowenBao** [reply]
Thanks Kunal, I'm happy to contribute to this, will be sending in a pr soon.

**Comment by kunal-vaishnavi** [comment]
[Here](https://github.com/microsoft/onnxruntime-genai/pull/1447) is the PR for bfloat16 support in the model builder.
